modules/beopBidAdapter.js

```diff
@@ -1,153 +1,214 @@
+  triggerPixel,
+  buildUrl,
+  logInfo,
+  getValue,
+  getBidIdParameter,
+} from "../src/utils.js";
+import { getRefererInfo } from "../src/refererDetection.js";
+import { registerBidder } from "../src/adapters/bidderFactory.js";
+import { config } from "../src/config.js";
+const BIDDER_CODE = "beop";
+const ENDPOINT_URL = "https://hb.beop.io/bid";
 const TCF_VENDOR_ID = 666;
 
-const validIdRegExp = /^[0-9a-fA-F]{24}$/
+const validIdRegExp = /^[0-9a-fA-F]{24}$/;
 
 export const spec = {
   code: BIDDER_CODE,
   gvlid: TCF_VENDOR_ID,
-  aliases: ['bp'],
+  aliases: ["bp"],
   /**
-    * Test if the bid request is valid.
-    *
-    * @param {bid} : The Bid params
-    * @return boolean true if the bid request is valid (aka contains a valid accountId or networkId and is open for BANNER), false otherwise.
-    */
-  isBidRequestValid: function(bid) {
+   * Test if the bid request is valid.
+   *
+   * @param {bid} : The Bid params
+   * @return boolean true if the bid request is valid (aka contains a valid accountId or networkId and is open for BANNER), false otherwise.
+   */
+  isBidRequestValid: function (bid) {
     const id = bid.params.accountId || bid.params.networkId;
-    if (id === null || typeof id === 'undefined') {
-      return false
+    if (id === null || typeof id === "undefined") {
+      return false;
     }
     if (!validIdRegExp.test(id)) {
-      return false
+      return false;
     }
-    return bid.mediaTypes.banner !== null && typeof bid.mediaTypes.banner !== 'undefined';
+    return (
+      bid.mediaTypes.banner !== null &&
+      typeof bid.mediaTypes.banner !== "undefined"
+    );
   },
   /**
-    * Create a BeOp server request from a list of BidRequest
-    *
-    * @param {validBidRequests[], ...} : The array of validated bidRequests
-    * @param {... , bidderRequest} : Common params for each bidRequests
-    * @return ServerRequest Info describing the request to the BeOp's server
-    */
-  buildRequests: function(validBidRequests, bidderRequest) {
+   * Create a BeOp server request from a list of BidRequest
+   *
+   * @param {validBidRequests[], ...} : The array of validated bidRequests
+   * @param {... , bidderRequest} : Common params for each bidRequests
+   * @return ServerRequest Info describing the request to the BeOp's server
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
     const slots = validBidRequests.map(beOpRequestSlotsMaker);
     const pageUrl = getPageUrl(bidderRequest.refererInfo, window);
     const gdpr = bidderRequest.gdprConsent;
     const firstSlot = slots[0];
+    const kwdsFromRequest = bidderRequest.ortb2?.site?.keywords;
+    let keywords = [];
+    if (kwdsFromRequest) {
+      if (isArray(kwdsFromRequest)) {
+        keywords = kwdsFromRequest;
+      } else {
+        if (kwdsFromRequest.indexOf(",") != -1) {
+          keywords = kwdsFromRequest.split(",");
+        } else {
+          keywords.push(kwdsFromRequest);
+        }
+      }
+    }
+
```

Comment:
If you want me to nitpick, you could write 
```
else if (…) {
  …
  …
} else {
  …
}
```

instead of  
```
else {
  if (…) {
    …
    …
  } else {
    …
  }
}
```